### DISCUSSION TITLE
Generated Mac app does not have Python in the Framework directory

### DISCUSSION THREAD

**Discussion by strickce** [pyinstaller/pyinstaller#8381] [Help]
Using a custom spec file, I can generate an exec and it runs perfectly, but with a console window.  I also specify an app, and it builds it with my ICON.  But the app does not run.  I tried running the exec located in the app contents, and it gives me error messages that indicate that there is no Frameworks/Python directory.  I am running pyinstaller 6.4.0, Python 3.11.4, on Ventura 13.6.5.
Probably something simple I missed, but I'm at a loss.  Appreciate any help.

Here is my spec file:

# -*- mode: python ; coding: utf-8 -*-
addedFiles= [ 
    ('snooper_mac.ppn', './pvporcupine/resources/keyword_files/mac'),
    ('snoopy_mac.ppn', './pvporcupine/resources/keyword_files/mac'),
    ('.venv/lib/python3.11/site-packages/pvporcupine/lib/mac/x86_64/libpv_porcupine.dylib' , './pvporcupine/lib/mac/x86_64'),
    ('.venv/lib/python3.11/site-packages/pvporcupine/lib/common/porcupine_params.pv' , './pvporcupine/lib/common'),
    ('.venv/lib/python3.11/site-packages/pvrhino/lib/mac/x86_64/libpv_rhino.dylib' , './pvrhino/lib/mac/x86_64'),
    ('.venv/lib/python3.11/site-packages/pvrhino/lib/common/rhino_params.pv' , './pvrhino/lib/common'),
    ('.venv/lib/python3.11/site-packages/pvrecorder/resources/scripts/platform.sh' , 'pvrecorder/resources/scripts'),
    ('.venv/lib/python3.11/site-packages/pvrecorder/lib/mac/x86_64/libpv_recorder.dylib' , './pvrecorder/lib/mac/x86_64')

]

a = Analysis(
    ['Findy.py'],
    pathex=[],
    binaries=[],
    datas=addedFiles,
    hiddenimports=[],
    hookspath=[],
    hooksconfig={},
    runtime_hooks=[],
    excludes=[],
    noarchive=False,
)
pyz = PYZ(a.pure)

exe = EXE(
    pyz,
    a.scripts,
    [],
    exclude_binaries=True,
    name='Findy',
    debug=False,
    bootloader_ignore_signals=False,
    strip=False,
    upx=True,
    console=True,
    disable_windowed_traceback=False,
    argv_emulation=False,
    target_arch=None,
    codesign_identity=None,
    entitlements_file=None,
    icon=['Findy.icns'],
)

coll = COLLECT(
    exe,
    a.binaries,
    a.datas,
    strip=False,
    upx=True,
    upx_exclude=[],
    name='Findy',
)

app = BUNDLE(exe,
         name='Findy.app',
         icon='Findy.icns',
         bundle_identifier='Findy')

The error from the console is:

/Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/MacOS/Findy ; exit;
~: /Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/MacOS/Findy ; exit;
[83962] Error loading Python lib '/Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/Frameworks/Python': dlopen: dlopen(/Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/Frameworks/Python, 0x000A): tried: '/Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/Frameworks/Python' (no such file), '/System/Volumes/Preboot/Cryptexes/OS/Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/Frameworks/Python' (no such file), '/Users/Charlie/Documents/AllProjects/Python/Projects/Findy/dist/Findy.app/Contents/Frameworks/Python' (no such file)
logout

**Comment by rokm** [comment]
Well, the error message is right. 

And that's because 

> `app = BUNDLE(exe,`

should be 
```
app = BUNDLE(coll,
```

instead.

**Comment by strickce** [comment]
Thank you so much!  I knew it would be something simple that I didn't know about.Sent from my iPadOn Mar 28, 2024, at 10:44 AM, Rok Mandeljc ***@***.***> wrote:﻿
Well, the error message is right.
And that's because

app = BUNDLE(exe,

should be
app = BUNDLE(coll,

instead.
